LanguageFeatures/Augmentation-libraries/augmenting_constructors_A11_t02_lib.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A non-redirecting generative constructor marked `augment` may:
+/// - Add or replace the body of the augmented constructor with a new body.
+/// ...
+///   - Initializer lists are not re-run, they have already executed and
+///     shouldn't be executed twice. The same goes for initializing formals and
+///     super parameters.
+///
+/// @description Checks that when `augmented()` is called in the body of an
+/// augmenting constructor initializing formals are not re-run.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A11_t02.dart';
+
+augment class C {
+  augment C(this.x) {
```

Comment:
We can't do that because an initializing formal is concerned with instance variable _initialization_, and initialization never invokes a setter, it directly stores a value in the storage location which was allocated for the given variable.

This is important if a subclass introduces a `set` setter (that is, a setter which isn't implicitly induced by a variable declaration), because we still need to initialize the variable storage, even in the case where nobody else can write to that location because the setter "captures" all attempts to write to that location, and the setter doesn't write to that location itself. So initialization really needs to have a bit of magic power beyond what we can express in code that runs after the initialization has completed.

Reply:
Rewritten. Simply check that final variable is initialized by the augmenting constructor and `augmented()` doesn't try to reinitialize it. I don't think that anything else can be done to tests it.

Reply:
Right, we probably can't detect whether the value provided to a parameter declared as `this.x` is stored in `x` once or twice: The operation that stores a value in a variable does not have any side effects, and we can't change the value of the parameter declared as `this.x` in the initializer list (it is implicitly final). Otherwise, we could have assigned to it in an initializer list element:

```dart
// It works as long as we're assigning a parameter which isn't initializing.
class A {
  int i, j;
  Object? dummy;
  A(int i): this.i = i, dummy = (i = i + 1), j = i;
  toString() => 'A(i: $i, j: $j)';
}

// But not with an initializing formal.
class B {
  int i, j;
  Object? dummy;
  A(this.i): dummy = (
    i = // Error, can't assign to `i` which is implicitly `final`.
    i + 1), j = i;
  toString() => 'A(i: $i, j: $j)';
}
```